ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -48,10 +48,23 @@ def record_error(id, payload, stats: nil)
         end
 
         def record_success(id, stats: nil)
-          redis.pipelined do |pipeline|
+          deleted_count, requeued_count, _ = redis.pipelined do |pipeline|
             pipeline.hdel(key('error-reports'), id.dup.force_encoding(Encoding::BINARY))
+            pipeline.hget(key('requeues-count'), id.b)
             record_stats(stats, pipeline: pipeline)
           end
+          record_flaky(id) if deleted_count == 1 || requeued_count.to_i > 0
```

Comment:
[nit] `deleted_count == 1` looks more like we deleted 1 element but `1` is, I believe` the exit code here.

So something like this might be more accurate

```suggestion
          SUCCESS_EXIT_CODE = 1
          record_flaky(id) if prev_error_reported == SUCCESS_EXIT_CODE || requeued_count.to_i > 0
```

Reply:
Based on the [redis doc](https://redis.io/commands/hdel/), it is returning number of fields removed. 

> Integer reply: the number of fields that were removed from the hash, not including specified but non existing fields.